src/custom/state/claim/updater.tsx

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from 'react'
+import { useBlockNumber } from '../application/hooks'
+import { SupportedChainId } from 'constants/chains'
+import { supportedChainId } from 'utils/supportedChainId'
+import { useActiveWeb3React } from 'hooks/web3'
+import { fetchClaims, useClaimDispatchers } from './hooks'
+
+const SUPPORTED_CLAIM_CHAINS = [SupportedChainId.MAINNET, SupportedChainId.XDAI]
+
+export default function Updater() {
+  const { account, chainId } = useActiveWeb3React()
+  const blockNumber = useBlockNumber()
+  const { setHasClaimsOnOtherChains } = useClaimDispatchers()
+
+  useEffect(() => {
+    if (!account) return
+
+    const supportedChain = supportedChainId(chainId)
+    SUPPORTED_CLAIM_CHAINS.forEach((chain) => {
+      if (supportedChain === chain) return
+      // Chain at this point is a supported chain
+      fetchClaims(account, chain)
```

Comment:
this doesn't account for the consumed claims.

Reply:
Yeah, you'll have to use `useUserAvailableClaims` hook here although it uses the connected chainId.
Will need to modify the hooks to take it as an optional parameter instead

Reply:
@anxolin https://github.com/gnosis/cowswap/pull/2320 @alfetopito